editor/editor_node.cpp

```diff
@@ -4954,6 +4959,37 @@ void EditorNode::_save_docks_to_config(Ref<ConfigFile> p_layout, const String &p
 		}
 	}
 
+	Dictionary floating_docks_dump;
+
+	for (Control *floating_dock : floating_docks) {
+		WindowWrapper *wrapper = Object::cast_to<WindowWrapper>(floating_dock);
```

Comment:
`floating_docks` can be changed to `Vector<WindowWrapper>`, so you wouldn't need to cast it.

Reply:
TBH, the whole dock code stuff should be rewritten to always wrap docks in WindowWrapper, like what I did in the Script and Shader editors.